regrid/regrid.py

```diff
@@ -0,0 +1,251 @@
+                        "vo": (None, "skip"),
+                        "taux": ("tauy", "U"),
+                        "tauy": (None, "skip"),
+                    }
+                    if var in variable_map.keys():
+                        var2, pos = variable_map[var]
+                    else:
+                        var2, pos = (None, "T")
+
+                # 3-dimensional data
+                is_3d = False
+                if coords.names[1] == "z_l" or coords.names[1] == "zl":
+                    dims = ["time", "lev", "lat", "lon"]
+                    is_3d = True
+                else:
+                    dims = ["time", "lat", "lon"]
+
+                if pos == "T":
+                    interp_out = self.rg_tt(ds_in[var].values)
+                    da_out = xr.DataArray(interp_out, dims=dims)
+                    da_out.attrs["long_name"] = ds_in[var].long_name
+                    da_out.attrs["units"] = ds_in[var].units
+                    ds_out.append(da_out.to_dataset(name=var))
+
+                if pos == "U":
+                    # interplate u and v to t-point, then rotate currents/winds to
+                    # earth relative before interpolation
+
+                    # interpolate to t-points
+                    interp_u = self.rg_ut(ds_in[var].values)
+                    interp_v = self.rg_ut(ds_in[var2].values)
+
+                    # rotate to earth-relative
+                    if is_3d:
+                        urot = np.zeros(np.shape(interp_u)[1:])
+                        vrot = np.zeros(np.shape(interp_u)[1:])
+                        for k in range(np.shape(interp_u)[1]):
+                            urot[k] = (
+                                interp_u[0, k, :, :] * self.ds_rot.cos_rot
+                                + interp_v[0, k, :, :] * self.ds_rot.sin_rot
+                            )
+                            vrot[k] = (
+                                interp_v[0, k, :, :] * self.ds_rot.cos_rot
+                                - interp_u[0, k, :, :] * self.ds_rot.sin_rot
+                            )
+                    else:
+                        urot = (
+                            interp_u[0, :, :] * self.ds_rot.cos_rot
+                            + interp_v[0, :, :] * self.ds_rot.sin_rot
+                        )
+                        vrot = (
+                            interp_v[0, :, :] * self.ds_rot.cos_rot
+                            - interp_u[0, :, :] * self.ds_rot.sin_rot
+                        )
+
+                    # interoplate
+                    uinterp_out = self.rg_tt(urot)
+                    vinterp_out = self.rg_tt(vrot)
+                    da_out = xr.DataArray(
+                        np.expand_dims(uinterp_out, 0),
+                        dims=dims,
+                    )
+                    da_out.attrs["long_name"] = ds_in[var].long_name
+                    da_out.attrs["units"] = ds_in[var].units

```

Comment:
More nitpick suggestions ...
```suggestion
                        attrs={
                            "long_name": ds_in[var].long_name,
                            "units": ds_in[var].units,
                        }
                    )
```